File: palin/python3/palin.py
testCase = input()
for t in range(int(testCase)):
    num = list(map(int, input()))
    n = len(num)
    
    all9 = all(x == 9 for x in num)
    if all9:
        print("1", end="")
        for i in range(1, n):
            print("0",end="")
        print("1")
    else:
        mid = int(n/2)
        indexOfEndOfLeftSide = mid - 1
        indexOfStartOfRightSide = mid + 1 if n % 2 else mid

        while indexOfEndOfLeftSide >= 0 and num[indexOfEndOfLeftSide] == num[indexOfStartOfRightSide]:
            indexOfEndOfLeftSide -= 1
            indexOfStartOfRightSide += 1
            
        leftSmaller = False
        if indexOfEndOfLeftSide < 0 or num[indexOfEndOfLeftSide] < num[indexOfStartOfRightSide]:
            leftSmaller = True

        while indexOfEndOfLeftSide >= 0:
            num[indexOfStartOfRightSide] = num[indexOfEndOfLeftSide]
            indexOfEndOfLeftSide -= 1
            indexOfStartOfRightSide += 1
        
        if leftSmaller:
            carry = 1
            indexOfEndOfLeftSide = mid - 1

            if n % 2 == 1:
                num[mid] += carry
                carry = int(num[mid] / 10)
                num[mid] %= 10
                indexOfStartOfRightSide = mid + 1
            else:
                indexOfStartOfRightSide = mid
            
            while indexOfEndOfLeftSide >= 0:
                num[indexOfEndOfLeftSide] += carry
                carry = int(num[indexOfEndOfLeftSide] / 10)
                num[indexOfEndOfLeftSide] %= 10
                num[indexOfStartOfRightSide] = num[indexOfEndOfLeftSide]
                indexOfEndOfLeftSide -= 1
                indexOfStartOfRightSide += 1

        for i in range(0, n):
            print(num[i], end="")
        print()

File: palin/csharp/palin.cs
using System;

public class Palin {
	public static void Main() {
		int testCase = int.Parse(Console.ReadLine());
        for (int t = 0; t < testCase; t++) {
            int[] num = Array.ConvertAll(Console.ReadLine().ToCharArray(), elem => int.Parse(elem.ToString()));
            int n = num.Length;

            bool all9 = Array.TrueForAll(num, elem => elem == 9);
            if(all9) {
                Console.Write('1');
                for (int i = 1; i < n; i++) {
                    Console.Write('0');
                }
                Console.WriteLine('1');
            } else {
                int mid = n / 2;
                int indexOfLeftSide = mid - 1;
                int indexOfRightSide = n % 2 == 0 ? mid : mid + 1;
                
                while(indexOfLeftSide >= 0 && num[indexOfLeftSide] == num[indexOfRightSide]) {
                    indexOfLeftSide--;
                    indexOfRightSide++;
                }

                bool leftSmaller = false;
                if(indexOfLeftSide < 0 || num[indexOfLeftSide] < num[indexOfRightSide])
                    leftSmaller = true;
                
                while(indexOfLeftSide >= 0) {
                    num[indexOfRightSide] = num[indexOfLeftSide];
                    indexOfLeftSide--;
                    indexOfRightSide++;
                }

                if(leftSmaller) {
                    int carry = 1;
                    indexOfLeftSide = mid - 1;

                    if(n % 2 == 1) {
                        num[mid] += carry;
                        carry = num[mid] / 10;
                        num[mid] %= 10;
                        indexOfRightSide = mid + 1;
                    } else
                        indexOfRightSide = mid;
                    
                    while(indexOfLeftSide >= 0) {
                        num[indexOfLeftSide] += carry;
                        carry = num[indexOfLeftSide] / 10;
                        num[indexOfLeftSide] %= 10;
                        num[indexOfRightSide] = num[indexOfLeftSide];
                        indexOfLeftSide--;
                        indexOfRightSide++;
                    }
                }
                for (int i = 0; i < n; i++) {
                    Console.Write(num[i]);
                }
                Console.WriteLine();
            }
        }
	}
}